validator/validator_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import "testing"
+
+func Test_getFieldKey(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		data      any
+		fieldName string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "struct_with_key_tag",
+			args: args{
+				data: &struct {
+					Name string `key:"super_name"`
```

Comment:
I just realized now why you used key tags in structures! That's a great idea. Kept it :)